### PR TITLE
Use buttonsArea, UI cleanup

### PR DESCRIPTION
**Pull request by irgolic**
To be merged after https://github.com/biolab/orange-widget-base/pull/130 is merged and released.

##### Description of changes
This PR moves some auto_commit buttons, and data projection controls, from controlArea to buttonsArea. With the widget-base PR, this is displayed below a scrollable controlArea, making it always visible.

Also, widget UIs are redesigned to be more consistent. All widgets except based only in mainArea were moved to controlArea (except Venn Diagram).

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by irgolic** [comment]
I only went over widgets who explicitly override `buttons_area_orientation`. However, that option defaulted to `Qt.Horizontal`, so I'll take a closer look to see if there's other widgets which should be fixed. The way biolab/orange-widget-base#102 deprecates the GUI layout setting though, it should be fine.

Some mini issues:

OWSilhouettePlot, OWRank: why is controlArea slightly smaller in width than buttonsArea (on Mac)? When I try to debug this by setting a `'background: red;'` stylesheet, the problem disappears. What a heisenbug.

OWContinuize: After moving it down, the 'Apply Automatically' button is gigantic. Moving it below the options is definitely better, but its width needs to be constrained. Perhaps a max width should be set if `want_main_area` is False?

**Comment by irgolic** [comment]
Due to https://github.com/biolab/orange-widget-base/pull/127, this will need to be adjusted to retain the `gui.rubber`s in the controlArea.

File: Orange/widgets/data/owpreprocess.py [hunk 1]
```diff
@@ -1142,8 +1154,8 @@ def mimeData(indexlist):
         self.mainArea.layout().addWidget(self.scroll_area)
         self.flow_view.installEventFilter(self)
 
-        box = gui.vBox(self.controlArea, "Output")
-        gui.auto_apply(box, self, "autocommit", box=False)
+        # autocommit is not in buttonsArea to allow list view stretch
```
Comment (markotoplak) [inline Orange/widgets/data/owpreprocess.py:1157]: Is this comment still relevant?

File: Orange/widgets/data/owpreprocess.py [hunk 2]
```diff
@@ -1105,8 +1105,20 @@ def mimeData(indexlist):
         self.preprocessors.mimeData = mimeData
 
         box = gui.vBox(self.controlArea, "Preprocessors")
+        gui.rubber(self.controlArea)
 
-        self.preprocessorsView = view = QListView(
+        class ListView(QListView):
```
Comment (markotoplak) [inline Orange/widgets/data/owpreprocess.py:1110]: Why is this needed?

File: Orange/widgets/gui.py [hunk 3]
```diff
@@ -34,6 +34,7 @@
     CalendarWidgetWithTime, DateTimeEditWCalendarTime,
     ControlGetter, VerticalScrollArea, ProgressBar,
     ControlledCallback, ControlledCallFront, ValueCallback, connectControl,
+    is_macstyle
```
Comment (markotoplak) [inline Orange/widgets/gui.py:37]: Isn't `Orange/widget/gui.py` only for backward compatibility and Orange-specific additions? If so, the new `is_macstyle` from `widget-base` should not be added here.

File: Orange/widgets/model/owloadmodel.py [hunk 4]
```diff
@@ -30,14 +30,15 @@ class Error(widget.OWWidget.Error):
...
+    buttons_area_orientation = None
     resizing_enabled = False
 
     def __init__(self):
         super().__init__()
         RecentPathsWComboMixin.__init__(self)
         self.loaded_file = ""
 
-        vbox = gui.vBox(self.controlArea, "File", addSpace=True)
+        vbox = gui.vBox(self.controlArea, "File", addSpace=False)
```
Comment (markotoplak) [inline Orange/widgets/model/owloadmodel.py:41]: `addSpace` was deprecated thus all uses need to be removed.

File: Orange/widgets/model/owrandomforest.py [hunk 5]
```diff
@@ -52,12 +52,14 @@ def add_main_layout(self):
...
-            callback=self.settings_changed)
+            callback=self.settings_changed,
+            attribute=Qt.WA_LayoutUsesWidgetRect)
         self.weights = gui.checkBox(
             box, self,
             "class_weight", label="Balance class distribution",
             callback=self.settings_changed,
-            tooltip="Weigh classes inversely proportional to their frequencies."
+            tooltip="Weigh classes inversely proportional to their frequencies.",
+            attribute=Qt.WA_LayoutUsesWidgetRect
```
Comment (markotoplak) [inline Orange/widgets/model/owrandomforest.py:62]: Could you also mention the `WA_LayoutUsesWidgetRect` trick in the style guide?

**Comment by markotoplak** [comment]
@irgolic, thank you, looks good!

I skimmed went through the changes and I have some (very) minor comments.
